include/dev/dfi/packet.h

```diff
@@ -572,9 +580,119 @@ wddr_return_t create_cke_packet_sequence(dfi_tx_packet_buffer_t *buffer,
+ *
+ * @return      void.
+ */
+void fill_address_packet(packet_item_t *packet,
+                         packet_group_info_t *addr_info,
+                         uint8_t cycles_per_packet,
+                         command_t *command,
+                         uint16_t phase_length);
+
+/**
+ * @brief   Fill RDDATA Packet
+ *
+ * @details Fills correct RDDATA phases for the specified packet group.
+ *
+ * @param[in]   packet              pointer to packet to fill.
+ * @param[in]   rddata_info         pointer to packet group info structure.
+ * @param[in]   cs                  chipselect value.
+ * @param[in]   cycles_per_packet   number of DRAM cycles per packet
+ *
+ * @return      void.
+ */
+void fill_rddata_packet(packet_item_t *packet,
+                        packet_group_info_t *rddata_info,
+                        chipselect_t cs,
+                        uint8_t cycles_per_packet);
+
+/**
+ * @brief   Fill WRDATA_EN Packet
+ *
+ * @details Fills correct WRDATA_EN phases for the specified packet group.
+ *
+ * @param[in]   packet              pointer to packet to fill.
+ * @param[in]   wrdata_en_info      pointer to packet group info structure.
+ * @param[in]   cycles_per_packet   number of DRAM cycles per packet
+ *
+ * @return      void.
+ */
+void fill_wrdata_en_packet(packet_item_t *packet,
+                           packet_group_info_t *wrdata_en_info,
+                           uint8_t cycles_per_packet);
+
+/**
+ * @brief   Fill WRDATA Packet
+ *
+ * @details Fills correct WRDATA phases for the specified packet group.
+ *
+ * @param[in]   packet              pointer to packet to fill.
+ * @param[in]   wrdata_info         pointer to packet group info structure.
+ * @param[in]   cs                  chipselect value.
+ * @param[in]   data                pointer to command data structure to send.
+ * @param[in]   cycles_per_packet   number of DRAM cycles per packet
+ * @param[in]   phase_length        expected total phases to be populated.
+ *
+ * @return      void.
+ */
+void fill_wrdata_packet(packet_item_t *packet,
+                        packet_group_info_t *wrdata_info,
+                        chipselect_t cs,
+                        command_data_t *data,
+                        uint8_t cycles_per_packet,
+                        uint16_t phase_length);
+
+/**
+ * @brief   Create Address Packet Sequence
+ *
+ * @details Creates a sequence of packets for sending commands to DRAM
+ *
+ * @param[in]   buffer          pointer to packet buffer to save packet to.
+ * @param[in]   ratio           ratio of DFI CK to DRAM CK.
+ * @param[in]   command         pointer to command structure to send.
+ * @param[in]   time_offset     time offset from previous packet that the packet
+ *
+ * @return      returns whether packet added to the packet buffer.
+ * @retval      WDDR_SUCCESS if added.
+ * @retval      WDDR_ERROR otherwise.
+ */
+wddr_return_t create_address_packet_sequence(dfi_tx_packet_buffer_t *buffer,
```

Comment:
Shouldn't this be in dram device?

Reply:
It looks like it can be a static function in dram device.

Reply:
Moved this and create_mrw_packet_sequence into dram device as static functions.